Storage/SQL/SQLiteHistory.swift

```diff
@@ -304,25 +545,8 @@ extension SQLiteHistory: BrowserHistory {
         }
     }
 
-    fileprivate func updateTopSitesCacheQuery() -> (String, Args?) {
-        let limit = Int(prefs.intForKey(PrefsKeys.KeyTopSitesCacheSize) ?? TopSiteCacheSize)
-
-        let (whereData, groupBy) = self.topSiteClauses()
-        let (query, args) = self.filteredSitesByFrecencyQueryWithHistoryLimit(limit, bookmarksLimit: 0, groupClause: groupBy, whereData: whereData)
-
-        // We must project, because we get bookmarks in these results.
-        let insertQuery = [
-            "INSERT INTO \(TableCachedTopSites)",
-            "SELECT historyID, url, title, guid, domain_id, domain,",
-            "localVisitDate, remoteVisitDate, localVisitCount, remoteVisitCount,",
-            "iconID, iconURL, iconDate, iconWidth, frecencies",
-            "FROM (", query, ")"
-            ].joined(separator: " ")
-        return (insertQuery, args)
-    }
-
     public func refreshTopSitesQuery() -> [(String, Args?)] {
```

Comment:
This should not be public; it's allowing a computed query to escape the situation in which it's valid. (Yes, that means that `SQLiteHistoryRecommendations` might need a small amount of thought.)

Reply:
The query was computed in the pre-PR code no?

Reply:
I'm not blaming you — it's not changed in your PR! — just making an observation!

Reply:
np, ideally we would have a "not the PR's fault" tag/acronym for review comments. Can be very different streams of thought for me depending on that.